Bs7 citations

stumbo opened the pull request:
Adds a citation button that brings up a modal window showing both the formatted citation and html text representing the formatted citation.  Allows user to copy either option.

Issue [2349](https://github.com/Interlisp/medley/issues/2349) has several screenshots.

stumbo committed:
Initial attempt to build citation modal dialog box.

Citation link, generates modal dialog box whose content is the
results of a call to Zotero to get the citation.  Currently
using apa citation, there are many other options...

stumbo committed:
Update Citation modal to include both rendered text and html

Allow the user to copy either the formatted 'pretty' text or
the raw html that creates the entry.

stumbo committed:
Fix reference to prettier js files in script.

stumbo committed:
Convert 'View on Zotero' and 'Citation' into buttons.

Move citations button to bottom of page with Zotero button.

stumbo committed:
Move Zotero group id from hardcoded to a site parameter

pamoroso reviewed (approved):
It works with no issues for me with Firefox and Chrome on Linux.

MattHeffron commented on layouts/bibliography/single.html:
```diff
@@ -209,13 +210,215 @@ <h1>{{ .Title }}</h1>
+  {{- end -}}
+
+        <!-- Citation -->
+    {{/* Build Zotero API params (group id + item key + style) */}}
```
The lines that are only Hugo operations should have {{- & -}} delimiters so whitespace between them is stripped from the output. (Generally, no functional effect, but helps with readability if looking at "page source" in the browser. Also, smaller pages.

MattHeffron commented on layouts/bibliography/single.html:
```diff
@@ -209,13 +210,215 @@ <h1>{{ .Title }}</h1>
+    <div class="citation-actions">
+      <button type="button" id="copyCitation">Copy Text</button>
+      <button type="button" id="copyCitationHtml">Copy HTML</button>
+      <button type="button" class="citation-close">Close</button>
```
This "Close" is a larger font, and no border, compared to the two buttons adjacent.

stumbo replied:
Fixed in the update.

stumbo committed:
Fix button styles and clean up Hugo operation syntax.

In the process I discovered a couple minor errors in the
layout of the bibliography and fixed them.
